pkg/commands/status.go

```diff
@@ -212,94 +250,50 @@ func NewCompStatusCommand(c types.Args, ioStreams cmdutil.IOStreams) *cobra.Comm
 }
 
 func printComponentStatus(ctx context.Context, c client.Client, ioStreams cmdutil.IOStreams, compName, appName string, env *types.EnvMeta) error {
-	ioStreams.Infof("Showing status of Component %s deployed in Environment %s\n", compName, env.Name)
-	var app *application.Application
-	var err error
-	if appName != "" {
-		app, err = application.Load(env.Name, appName)
-	} else {
-		app, err = application.MatchAppByComp(env.Name, compName)
-	}
+	time.Sleep(15 * time.Second)
+	app, appConfig, err := getAppAndApplicationConfiguration(ctx, c, compName, appName, env)
 	if err != nil {
 		return err
 	}
-
-	var appConfig v1alpha2.ApplicationConfiguration
-	if err = c.Get(ctx, client.ObjectKey{Namespace: env.Namespace, Name: app.Name}, &appConfig); err != nil {
-		return err
+	if app == nil || appConfig == nil {
+		return errors.New(ErrNotLoadAppConfig)
 	}
 
-	var wlStatus v1alpha2.WorkloadStatus // very important
-	for _, v := range appConfig.Status.Workloads {
-		if v.ComponentName == compName {
-			wlStatus = v
-			break
-		}
-	}
-	if wlStatus.ComponentName == "" {
-		//TODO(roywang) cannot get workload instance
-		//TODO(roywang) if appConfig reconcile condition is false, then output err msg
-		ioStreams.Infof("\nComponent is not created yet.")
+	wlStatus, foundWlStatus := getWorkloadStatusFromAppConfig(appConfig, compName)
+	if !foundWlStatus {
+		ioStreams.Info("\nThe component has not been initialized by oam-core-controller correctly.")
```

Comment:
> not been initialized by oam-core-controller correctly

Users won't understand what does it mean, can you give more concrete error information